streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -47,37 +47,43 @@
 
     private static final String REDUCE_NAME = "KTABLE-REDUCE-";
 
-    private final String userSpecifiedName;
+    private final String userProvidedRepartitionTopicName;
 
     private final Initializer<Long> countInitializer = () -> 0L;
 
     private final Aggregator<K, V, Long> countAdder = (aggKey, value, aggregate) -> aggregate + 1L;
 
     private final Aggregator<K, V, Long> countSubtractor = (aggKey, value, aggregate) -> aggregate - 1L;
 
+    private StreamsGraphNode repartitionGraphNode;
+
     KGroupedTableImpl(final InternalStreamsBuilder builder,
                       final String name,
                       final Set<String> sourceNodes,
                       final GroupedInternal<K, V> groupedInternal,
                       final StreamsGraphNode streamsGraphNode) {
         super(name, groupedInternal.keySerde(), groupedInternal.valueSerde(), sourceNodes, streamsGraphNode, builder);
 
-        this.userSpecifiedName = groupedInternal.name();
+        this.userProvidedRepartitionTopicName = groupedInternal.name();
     }
 
     private <T> KTable<K, T> doAggregate(final ProcessorSupplier<K, Change<V>> aggregateSupplier,
                                          final String functionName,
                                          final MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materialized) {
+
         final String sinkName = builder.newProcessorName(KStreamImpl.SINK_NAME);
         final String sourceName = builder.newProcessorName(KStreamImpl.SOURCE_NAME);
         final String funcName = builder.newProcessorName(functionName);
-        final String repartitionTopic = (userSpecifiedName != null ? userSpecifiedName : materialized.storeName())
+        final String repartitionTopic = (userProvidedRepartitionTopicName != null ? userProvidedRepartitionTopicName : materialized.storeName())
             + KStreamImpl.REPARTITION_TOPIC_SUFFIX;
 
-        final StreamsGraphNode repartitionNode = createRepartitionNode(sinkName, sourceName, repartitionTopic);
+        if (repartitionGraphNode == null || userProvidedRepartitionTopicName == null) {
```

Comment:
On the first invocation, the `repartitionNode` is always null, so we'll build it and set the variable. But on subsequent calls, only if the user has provided a name for the repartition topic will we re-use the repartition graph node.

This maintains compatibility as users not naming the repartition topic will get the same topology and re-using the `KGroupedTable` results in an `InvalidToplogyException` when building the topology